### PR TITLE
{vis} [intel/2015a] Gtkmm and deps (pango, GLib, cairo, Atkmm) (REVIEW)

### PR DESCRIPTION
**Pull request by JensTimmerman**

### CODE REVIEW

**Comment by boegel** [comment]
Please make Jenkins happy first?

**Comment by JensTimmerman** [comment]
@boegel jenkins is happy

**Comment by JensTimmerman** [comment]
@boegel ping

File: easybuild/easyconfigs/g/GLib/GLib-2.41.2-intel-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name = 'GLib'
+version = '2.41.2'
+easyblock = 'ConfigureMake'
+
+homepage = 'http://www.gtk.org/'
+description = """GLib is one of the base libraries of the GTK+ project"""
+toolchain = {'name': 'intel', 'version': '2015a'} # tweaked by EasyBuild (was: {'name': 'foss', 'version': '2014b'})
```
Comment (boegel) [inline easybuild/easyconfigs/g/GLib/GLib-2.41.2-intel-2015a.eb]: please fix this

File: easybuild/easyconfigs/g/GLibmm/GLibmm-2.41.2-intel-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+description = """GLib is one of the base libraries of the GTK+ project"""
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+toolchainopts = {'optarch': True, 'pic': True}
+
+source_urls = ['http://ftp.gnome.org/pub/gnome/sources/glibmm/%(version_major_minor)s/']
+sources = ['%(namelower)s-%(version)s.tar.xz']
+
+dependencies = [
+    ('GLib', '%(version)s'),
```
Comment (boegel) [inline easybuild/easyconfigs/g/GLibmm/GLibmm-2.41.2-intel-2015a.eb]: just use `version`?

File: easybuild/easyconfigs/g/gdk-pixbuf/gdk-pixbuf-2.31.4-intel-2015a.eb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+easyblock = 'ConfigureMake'
+name = 'gdk-pixbuf'
```
Comment (boegel) [inline easybuild/easyconfigs/g/gdk-pixbuf/gdk-pixbuf-2.31.4-intel-2015a.eb]: insert empty line

File: easybuild/easyconfigs/h/HarfBuzz/HarfBuzz-0.9.41-intel-2015a.eb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+homepage = 'http://www.freedesktop.org/wiki/Software/HarfBuzz'
+description = """HarfBuzz is an OpenType text shaping engine."""
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+source_urls = ['http://www.freedesktop.org/software/harfbuzz/release/']
+sources = [SOURCELOWER_TAR_BZ2]
+
+patches = ['HarfBuzz_no_symbolic.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/h/HarfBuzz/HarfBuzz-0.9.41-intel-2015a.eb]: please version this to `HarfBuzz-0.9.41_no_symbolic.patch`

File: easybuild/easyconfigs/l/libsigc++/libsigc++-2.4.1-intel-2015a.eb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+source_urls = ['http://ftp.gnome.org/pub/gnome/sources/%(name)s/%(version_major_minor)s/']
+sources = ['%(namelower)s-%(version)s.tar.xz']
+
+sanity_check_paths = {
+	'files': ['lib/libsigc-2.0.so'],
+        'dirs': [],
+}
+
+moduleclass = 'devel'
+
```
Comment (boegel) [inline easybuild/easyconfigs/l/libsigc++/libsigc++-2.4.1-intel-2015a.eb]: drop empty line?

File: easybuild/easyconfigs/p/Pango/Pango-1.37.1-intel-2015a.eb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+easyblock = 'ConfigureMake'
+name = 'Pango'
```
Comment (boegel) [inline easybuild/easyconfigs/p/Pango/Pango-1.37.1-intel-2015a.eb]: insert empty line?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 11 out of 13 (13 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e5ca5be15aec86d4b4b4 for a full test report.

**Comment by boegel** [comment]
@JensTimmerman: issues with the gtk+ build, see test report...

**Comment by JensTimmerman** [comment]
I believe this is because you changed some of the dependencies of the dependencies in your previous pull request and didn't sync them up here yet

```

libpng/1.6.17-intel-2015a(11):ERROR:150: Module 'libpng/1.6.17-intel-2015a' conflicts with the currently loaded module(s) 'libpng/1.6.16-intel-2015a'
libpng/1.6.17-intel-2015a(11):ERROR:102: Tcl command execution failed: conflict libpng

```

File: easybuild/easyconfigs/g/gdk-pixbuf/gdk-pixbuf-2.31.4-intel-2015a.eb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+source_urls = ['http://ftp.gnome.org/pub/GNOME/sources/%(namelower)s/%(version_major_minor)s']
+sources = [SOURCELOWER_TAR_XZ]
+
+dependencies = [
+    ('GLib', '2.41.2'),
+    ('libjpeg-turbo', '1.4.0'),
+    ('libpng', '1.6.16'),
```
Comment (JensTimmerman) [inline easybuild/easyconfigs/g/gdk-pixbuf/gdk-pixbuf-2.31.4-intel-2015a.eb]: should be 1.6.17

**Comment by JensTimmerman** [comment]
This is why I generally don't bump versions of dependencies within the same toolchain, it becomes a lot harder to start combining modules within the same toolchain if some of them depend on libpng 1.6.16 and others on 1.6.17.

**Comment by boegel** [comment]
The easyconfigs unit tests are supposed to catch this, but they got broken, see #1870

**Comment by boegel** [comment]
@JensTimmerman: this is also why we need support for specifying `'libpng', '1.6.*'` as a dependency, but until someone does the work, we'll keep suffering

**Comment by boegel** [comment]
@JensTimmerman: please merge in https://github.com/JensTimmerman/easybuild-easyconfigs/pull/24 to fix the conflict detection

Once that fix is confirmed here, we can follow up with fixing the conflicts in this PR.

**Comment by boegel** [comment]
@JensTimmerman: please fix the conflicts? let's get this one out of the way...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 13 out of 13 (13 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/6330f537eb24f6baa8b6 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @JensTimmerman!

**Pull request merged**